lib/mayaUsd/fileio/translators/translatorMaterial.cpp

```diff
@@ -89,10 +143,87 @@ MObject UsdMayaTranslatorMaterial::Read(
     return shadingEngine;
 }
 
+namespace {
+using _UVBindings = std::map<TfToken, TfToken>;
+}
+
+static _UVBindings
+_GetUVBindingsFromMaterial(const UsdShadeMaterial& material, UsdMayaPrimReaderContext* context)
+{
+    _UVBindings retVal;
+
+    if (!material || !context) {
+        return retVal;
+    }
+    const bool isMergeable = _IsMergeableMaterial(material);
+    // Find out the nodes requiring mapping. This code has deep knowledge of how the mappings are
+    // exported. See the _UVMappingManager class in ..\shading\shadingModeExporterContext.cpp for
+    // details.
+    for (const UsdShadeInput& input : material.GetInputs()) {
+        const UsdAttribute&      usdAttr = input.GetAttr();
+        std::vector<std::string> splitName = usdAttr.SplitName();
+        if (splitName.size() != 3 || splitName[2] != _tokens->varname.GetString()) {
+            continue;
+        }
+        VtValue val;
+        usdAttr.Get(&val);
+        if (!val.IsHolding<TfToken>()) {
+            continue;
+        }
+        SdfPath nodePath = isMergeable
+            ? material.GetBaseMaterial().GetPath().AppendChild(TfToken(splitName[1]))
+            : material.GetPath().AppendChild(TfToken(splitName[1]));
+        MObject           mayaNode = context->GetMayaNode(nodePath, false);
+        MStatus           status;
+        MFnDependencyNode depFn(mayaNode, &status);
+        if (!status) {
+            continue;
+        }
+        retVal[val.UncheckedGet<TfToken>()] = TfToken(depFn.name().asChar());
+    }
+
+    return retVal;
+}
+
+static void _BindUVs(const MDagPath& shapeDagPath, const _UVBindings& uvBindings)
+{
+    if (uvBindings.empty()) {
+        return;
+    }
+
+    MStatus status;
+    MFnMesh meshFn(shapeDagPath.node(), &status);
+    if (!status) {
+        return;
+    }
+
+    MStringArray uvSets;
+    meshFn.getUVSetNames(uvSets);
+
+    // We explicitly skip uvSet[0] since it is the default in Maya and does not require explicit
+    // linking:
+    for (unsigned int uvSetIndex = 1; uvSetIndex < uvSets.length(); uvSetIndex++) {
+        TfToken                     uvSetName(uvSets[uvSetIndex].asChar());
+        _UVBindings::const_iterator iter = uvBindings.find(uvSetName);
+        if (iter == uvBindings.cend()) {
+            continue;
+        }
+        MString uvLinkCommand("uvLink -make -uvs \"");
+        uvLinkCommand += shapeDagPath.fullPathName();
+        uvLinkCommand += ".uvSet[";
+        uvLinkCommand += uvSetIndex;
+        uvLinkCommand += "].uvSetName\" -t \"";
+        uvLinkCommand += iter->second.GetText(); // texture name
+        uvLinkCommand += "\";";
+        status = MGlobal::executeCommand(uvLinkCommand);
```

Comment:
Yikes. I take it there's no C++ API for this?

Reply:
There is no direct C++ API to do this. The command has some interesting logic that would have to be reproduced. This includes navigation from the texture node to the placement node, and the decision to add or reuse a uvChooser depending on the inputs of the placement node.